doc/modules/decomposition.rst

```diff
@@ -278,6 +257,73 @@ factorization, while larger values shrink many coefficients to zero.
      R. Jenatton, G. Obozinski, F. Bach, 2009
 
 
+.. _kernel_PCA:
+
+Kernel Principal Component Analysis (kPCA)
+==========================================
+
+Exact Kernel PCA
+----------------
+
+:class:`KernelPCA` is an extension of PCA which achieves non-linear
+dimensionality reduction through the use of kernels (see :ref:`metrics`). It
+has many applications including denoising, compression and structured
+prediction (kernel dependency estimation). :class:`KernelPCA` supports both
+``transform`` and ``inverse_transform``.
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca.html
+    :align: center
+    :scale: 75%
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca.py`
+
+.. _kPCA_Solvers:
+
+Choice of solver for Kernel PCA
+-------------------------------
+
+In :class:`KernelPCA`, the number of components found is equal to the number of
+samples. Many real-world datasets have large number of samples! In these cases
+finding *all* the components with a full kPCA is a waste of computation time,
+as data is mostly described by the first few components
+(e.g. ``n_components<=100``).
+
+The optional parameter ``eigen_solver='randomized'`` can be used to
+*significantly* reduce the computation time when the number of requested
+``n_components`` is small compared with the number of samples.
+This is done automatically by default if you do not specify any
+``eigen_solver`` and if the required ``n_components`` is less than 80% of
+the number of samples, see :class:`KernelPCA` for details.
+
+For example the figure below shows the same "circles" dataset (top left)
+reconstructed with various approximations (number of components decreases from
+2000 - full - to 4). You can compare the execution times vs. the reconstruction
+quality :
+
+.. figure:: ../auto_examples/decomposition/images/sphx_glr_plot_kernel_pca_approximate_001.png
+    :target: ../auto_examples/decomposition/plot_kernel_pca_approximate.html
+    :align: center
+    :scale: 75%
+
+The time complexity of the randomized :class:`KernelPCA` is
+:math:`O(n_{\mathrm{samples}}^2 \cdot n_{\mathrm{components}})`
+instead of :math:`O(n_{\mathrm{samples}}^3)` for the exact method
+implemented with ``eigen_solver='dense'``.
+
+The memory footprint of randomized :class:`KernelPCA` is also proportional to
+:math:`2 \cdot n_{\mathrm{samples}} \cdot n_{\mathrm{components}}` instead of
+:math:`n_{\mathrm{samples}}^2` for the exact method.
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_decomposition_plot_kernel_pca_approximate.py`
+
+Note: this technique is the same than in :ref:`RandomizedPCA`.
```

Comment:
same as

And please put this above the example section

Reply:
done